### DISCUSSION TITLE
How to manually attach custom part to a ContentItem?

### DISCUSSION THREAD

**Discussion by MikeAlhayek** [OrchardCMS/OrchardCore#10824] [Q&A] [answered]
I have two modules/features "Events" and "Teams". The events module provides a content-type called `Event` whereas the Teams feature provides content part called `MultipleTeamPart`. The `MultipleTeamPart` provided a way for the user to assign an event to a specific team.

When both featured "Events and Teams" are enabled, I want to always attach `MultipleTeamPart` to the `Event` content item during edit.  The `MultipleTeamPart` looks like this

```
public class MultipleTeamPart : ContentPart
{
    public ContentPickerField Team { get; set; }
}
```
The `MultipleTeamPart` part is registered like this

```
public class TeamsStartup : StartupBase
{
    public override void ConfigureServices(IServiceCollection services)
    {
        services.AddContentPart<MultipleTeamPart>()
                     .UseDisplayDriver<MultipleTeamPartDisplayDriver>();

        services.AddScoped<IDataMigration, TeamMigrations>();

       // other services
    }
}
```

In the Events module, I created the following display driver to weld the `MultipleTeamPart` to the event content item. 
*Note,* here `EventVisibilityPart` is already attached to `Event` content-type, so anytime that is displayed I would weld the  `MultipleTeamPart`

```
public class EventVisibilityPartWithTeamDisplayDriver : ContentPartDisplayDriver<EventVisibilityPart>
{
    public override IDisplayResult Edit(EventVisibilityPart part, BuildPartEditorContext context)
    {
        part.ContentItem.Weld<MultipleTeamPart>();

        return base.Edit(part, context);
    }
}
```
Finally, to use the `EventVisibilityPartWithTeamDisplayDriver` when both `Teams` and `Events` are enabled, I have the following startup class

```
[RequireFeatures("Events", "Teams")]
public class EventsWithTeamsStartup : StartupBase
{
    public override void ConfigureServices(IServiceCollection services)
    {
        services.AddScoped<IContentPartDisplayDriver, EventVisibilityPartWithTeamDisplayDriver>();
    }
}
```

However, the `MultipleTeamPart` does not show up on the UI even when the `EventVisibilityPartWithTeamDisplayDriver` is being called as expected.

I am guessing `MultipleTeamPart` is not showing up because I have not assigned `Location()` where to place it. My question, how can I assigned placement location for the `MultipleTeamPart` when I manually attach it as shown above?

I tried changing my driver to the the following 

```
public class EventVisibilityPartWithTeamDisplayDriver : ContentPartDisplayDriver<EventVisibilityPart>
{
    public override IDisplayResult Edit(EventVisibilityPart part, BuildPartEditorContext context)
    {
        part.ContentItem.Weld<MultipleTeamPart>();
        var m = part.ContentItem.As<MultipleTeamPart>();
        return View("MultipleTeamPart_Edit", m).Location("Content:10").OnGroup(context.GroupId);
    }
}
```

But I am not sure how will `MultipleTeamPart.Edit.cshtml` look like to render the `Team` field as the UI renders all the fields?

**Comment by ns8482e** [comment, marked as answer]
For `EventVisibilityPartWithTeamDisplayDriver` use `ContentDisplayDriver` and use `ViewModel` and read using `contentItem.As<>()`  and Weld/Merge in Update
